core/src/main/java/org/apache/druid/java/util/common/StringUtils.java

```diff
@@ -360,4 +360,93 @@ public static String encodeBase64String(byte[] input)
   {
     return BASE64_DECODER.decode(input);
   }
+
+  /**
+   * Returns the string left-padded with the string pad to a length of len characters.
+   * If str is longer than len, the return value is shortened to len characters. 
+   *
+   * @param base The base string to be padded
+   * @param len The length of padded string
+   * @param pad The pad string
+   * @return the string left-padded with pad to a lenght of len
+   */
+  public static String lpad(String base, Integer len, String pad)
+  {
+    if (len < 0) {
+      return null;
+    } else if (len == 0) {
+      return "";
+    }
+
+    char[] data = new char[len];
+    char[] baseChars = base.toCharArray();
```

Comment:
Is there a reason you copy `base` and `pad` to their own char arrays instead of just calling `charAt(index)` on the original strings?